numpyro/diagnostics.py

```diff
@@ -24,7 +24,7 @@
 ]
 
 
-def _compute_chain_variance_stats(x):
+def _compute_chain_variance_stats(x: np.ndarray) -> tuple[np.ndarray, np.ndarray]:
```

Comment:
All of these functions assume true numpy arrays, so this was the only way to make it work without adding many changes.